search-app/src/cmr/search/services/query_service.clj

```diff
@@ -106,35 +106,51 @@
     (assoc params :tag-data tag-data)
     params))
 
+(defn- make-concepts-tag-data
+  "Utility function for extracting tag data from params.
+
+  Note that the tag-data param has tag-key as its key, which could be any
+  arbitrary string. sanitize-params function traded data integrity for easier
+  processing, e.g. it converts \"tag1\" into :tag-1 and it doesn't work for
+  tag-keys. To work around this problem, we extract the tag-data param out
+  first, then add it back in after the sanitize-params call."
+  [params]
+  (or (:tag-data params) (:tag_data params)))
+
+(defn make-concepts-query
+  "Utility function for generating an elastic-ready query."
+  ([context concept-type params]
+    (->> params
+         (make-concepts-tag-data)
+         (make-concepts-query
+           context concept-type params)))
+  ([context concept-type params tag-data]
```

Comment:
Yeah, in fact the multi-arity here was added to support the original usage of this code (passing the `tag-data` parameter).

I've seen multi-arity functions used a ton for additional args. In fact, one of the most common patterns I've seen for Clojure multi-arity functions is for lower-arity to pass defaults to higher-arity.